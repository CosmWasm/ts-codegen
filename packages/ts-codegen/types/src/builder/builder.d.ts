import { RenderOptions } from "wasm-ast-types";
import { IBuilderPlugin } from '../plugins';
export interface TSBuilderInput {
    contracts: Array<ContractFile | string>;
    outPath: string;
    options?: TSBuilderOptions;
    plugins?: IBuilderPlugin[];
}
export interface BundleOptions {
    enabled?: boolean;
    scope?: string;
    bundleFile?: string;
    bundlePath?: string;
}
export declare type TSBuilderOptions = {
    bundle?: BundleOptions;
} & RenderOptions;
export declare type BuilderFileType = 'type' | 'client' | 'recoil' | 'react-query' | 'message-composer' | 'msg-builder' | 'plugin' | 'abstract-app';
export interface BuilderFile {
    type: BuilderFileType;
    pluginType?: string;
    contract: string;
    localname: string;
    filename: string;
}
export interface ContractFile {
    name: string;
    dir: string;
}
export declare class TSBuilder {
    contracts: Array<ContractFile | string>;
    outPath: string;
    options?: TSBuilderOptions;
    plugins: IBuilderPlugin[];
    protected files: BuilderFile[];
    loadDefaultPlugins(): void;
    constructor({ contracts, outPath, options, plugins }: TSBuilderInput);
    build(): Promise<void>;
    private process;
    private render;
    private after;
    bundle(): Promise<void>;
}
