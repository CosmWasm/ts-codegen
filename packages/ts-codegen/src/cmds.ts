import _create_boilerplate from './commands/create-boilerplate';
import _generate from './commands/generate';
import _install from './commands/install';
const Commands = {};
Commands['create-boilerplate'] = _create_boilerplate;
Commands['generate'] = _generate;
Commands['install'] = _install;

export { Commands };
