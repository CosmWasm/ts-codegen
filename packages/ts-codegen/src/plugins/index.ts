export * from "./plugin-base"
export * from "./client"
export * from "./react-query"
export * from "./recoil"
export * from "./message-builder"
export * from "./message-composer"
export * from "./types"