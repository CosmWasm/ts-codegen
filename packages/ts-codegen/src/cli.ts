import { prompt } from './utils/prompt';
import { Commands as commands } from './cmds';

const question = [
  {
    _: true,
    type: 'fuzzy',
    name: 'cmd',
    message: 'what do you want to do?',
    choices: Object.keys(commands)
  }
];

export const cli = async (argv) => {
  var { cmd } = await prompt(question, argv);
  if (typeof commands[cmd] === 'function') {
    await commands[cmd](argv);
  } else {
    console.log('command not found.');
  }
};
