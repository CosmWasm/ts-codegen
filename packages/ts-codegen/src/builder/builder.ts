import { RenderOptions, defaultOptions, RenderContext, ContractInfo, MessageComposerOptions} from "wasm-ast-types";

import { header } from '../utils/header';
import { join } from "path";
import { writeFileSync } from 'fs';
import { sync as mkdirp } from "mkdirp";

import { basename } from 'path';
import { readSchemas } from '../utils';
import { IBuilderPlugin } from '../plugins';

import deepmerge from 'deepmerge';
import { pascal } from "case";
import { createFileBundle, recursiveModuleBundle } from "../bundler";

import generate from '@babel/generator';
import * as t from '@babel/types';
import { ReactQueryPlugin } from "../plugins/react-query";
import { RecoilPlugin } from "../plugins/recoil";
import { MsgBuilderPlugin } from "../plugins/msg-builder";
import { MessageComposerPlugin } from "../plugins/message-composer";
import { ClientPlugin } from "../plugins/client";
import { TypesPlugin } from "../plugins/types";
import { AbstractAppPlugin } from '../plugins/abstract-app';

const defaultOpts: TSBuilderOptions = {
    bundle: {
        enabled: true,
        scope: 'contracts',
        bundleFile: 'bundle.ts'
    }
}

export interface TSBuilderInput {
    contracts: Array<ContractFile | string>;
    outPath: string;
    options?: TSBuilderOptions;
    plugins?: IBuilderPlugin[];
};

export interface BundleOptions {
    enabled?: boolean;
    scope?: string;
    bundleFile?: string;
    bundlePath?: string;
};

export type TSBuilderOptions = {
    bundle?: BundleOptions;
} & RenderOptions;

export type BuilderFileType = 'type' | 'client' | 'recoil' | 'react-query' | 'message-composer' | 'msg-builder' | 'plugin' | 'abstract-app';

export interface BuilderFile {
    type: BuilderFileType;
    pluginType?: string;
    contract: string;
    localname: string;
    filename: string;
};

export interface ContractFile {
    name: string;
    dir: string;
}

function getContract(contractOpt): ContractFile {
    if (typeof contractOpt === 'string') {
        const name = basename(contractOpt);
        const contractName = pascal(name);
        return {
            name: contractName,
            dir: contractOpt
        }
    }
    return {
        name: pascal(contractOpt.name),
        dir: contractOpt.dir
    };
}

export class TSBuilder {
    contracts: Array<ContractFile | string>;
    outPath: string;
    options?: TSBuilderOptions;
    plugins: IBuilderPlugin[] = [];

    protected files: BuilderFile[] = [];

    loadDefaultPlugins() {
        [].push.apply(this.plugins, [
            new TypesPlugin(this.options),
            new ClientPlugin(this.options),
            new AbstractAppPlugin(this.options),
            new MessageComposerPlugin(this.options),
            new ReactQueryPlugin(this.options),
            new RecoilPlugin(this.options),
            new MsgBuilderPlugin(this.options),
        ]);
    }

    constructor({ contracts, outPath, options, plugins }: TSBuilderInput) {
        this.contracts = contracts;
        this.outPath = outPath;
        this.options = deepmerge(
            deepmerge(
                defaultOptions,
                defaultOpts
            ),
            options ?? {}
        );

        this.loadDefaultPlugins();

        if (plugins && plugins.length) {
            [].push.apply(this.plugins, plugins);
        }
    }

    async build() {
        await this.process();
        await this.after();
    }

    // lifecycle functions
    private async process(){
        for (const contractOpt of this.contracts) {
            const contract = getContract(contractOpt);
            //resolve contract schema.
            const contractInfo = await readSchemas({
                schemaDir: contract.dir
            });

            //lifecycle and plugins.
            await this.render(contract.name, contractInfo);
        }
    }

    private async render(name: string, contractInfo: ContractInfo){
      for (const plugin of this.plugins) {
          let files = await plugin.render(name, contractInfo, this.outPath);
          if(files && files.length){
              [].push.apply(this.files, files);
          }
      }
    }

    private async after(){
        if (this.options.bundle.enabled) {
            this.bundle();
        }
    }

    async bundle() {

        const allFiles = this.files;

        const bundleFile = this.options.bundle.bundleFile;
        const bundlePath = join(
          this.options?.bundle?.bundlePath ?? this.outPath,
          bundleFile
        );
        const bundleVariables = {};
        const importPaths = [];

        allFiles.forEach(file => {
            createFileBundle(
                `${this.options.bundle.scope}.${file.contract}`,
                file.filename,
                bundlePath,
                importPaths,
                bundleVariables
            );

        });

        const ast = recursiveModuleBundle(bundleVariables);
        let code = generate(t.program(
            [
                ...importPaths,
                ...ast
            ]
        )).code;

        if(this.options?.bundle?.bundlePath){
            mkdirp(this.options?.bundle?.bundlePath);
        }

        mkdirp(this.outPath);

        if (code.trim() === '') code = 'export {};'

        writeFileSync(bundlePath, header + code);

    }
}
