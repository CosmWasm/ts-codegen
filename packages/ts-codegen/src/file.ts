#!/usr/bin/env node
import { prompt } from './utils/prompt';
import { cli } from './cli';
import { readFileSync } from 'fs';
const argv = require('minimist')(process.argv.slice(2));

const question = [
  {
    _: true,
    type: 'string',
    name: 'file',
    message: 'file'
  }
];

(async () => {
  const { file } = await prompt(question, argv);
  const argvv = JSON.parse(readFileSync(file, 'utf-8'));
  await cli(argvv);
})();
