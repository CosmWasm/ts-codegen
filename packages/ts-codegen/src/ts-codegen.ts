#!/usr/bin/env node
import { cli } from './cli';
var argv = require('minimist')(process.argv.slice(2));

(async () => {
  await cli(argv);
})();
