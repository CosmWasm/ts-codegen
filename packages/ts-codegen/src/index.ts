import { TSBuilder, TSBuilderInput } from './builder';

export { default as generateTypes } from './generators/types';
export { default as generateClient } from './generators/client';
export { default as generateMessageComposer } from './generators/message-composer';
export { default as generateMessageBuilder } from './generators/message-builder';
export { default as generateReactQuery } from './generators/react-query';
export { default as generateRecoil } from './generators/recoil';

export * from './utils';
export * from './builder';
export * from './bundler';
export * from './plugins';

export default async (input: TSBuilderInput) => {
    const builder = new TSBuilder(input);
    await builder.build();
};