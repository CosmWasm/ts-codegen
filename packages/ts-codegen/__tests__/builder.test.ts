import codegen from '../src/index';
import { TSBuilder } from '../src/builder';

const FIXTURE_DIR = __dirname + '/../../../__fixtures__';
const OUTPUT_DIR = __dirname + '/../../../__output__/builder';

it('options undefined', async () => {
    const outPath = OUTPUT_DIR + '/vectis/factory-opt';
    const schemaDir = FIXTURE_DIR + '/vectis/factory/';
    const builder = new TSBuilder({
        contracts: [
            schemaDir
        ],
        outPath,
        options: undefined
    });
    // @ts-ignore
    delete builder.contracts;
    // @ts-ignore
    delete builder.outPath;
    expect(builder).toMatchSnapshot();
});

it('options tsClient.enabled', async () => {
    const outPath = OUTPUT_DIR + '/vectis/factory-opt';
    const schemaDir = FIXTURE_DIR + '/vectis/factory/';
    const builder = new TSBuilder({
        contracts: [
            schemaDir
        ],
        outPath,
        options: {
            types: {
                enabled: true
            },
            client: {
                enabled: true
            },
            reactQuery: {
                enabled: true
            }
        }
    });
    // @ts-ignore
    delete builder.contracts;
    // @ts-ignore
    delete builder.outPath;
    expect(builder).toMatchSnapshot();
});

it('builder invoke', async () => {
    const outPath = OUTPUT_DIR + '/invoke/';
    const s = (str: string) => FIXTURE_DIR + str;
    const builder = new TSBuilder({
        contracts: [
            s('/vectis/factory'),
            s('/minter'),
            s('/daodao/cw-admin-factory'),
            s('/daodao/cw-code-id-registry'),
            {
                name: 'CwSingle',
                dir: s('/daodao/cw-proposal-single')
            }
        ],
        outPath,
        options: {
            types: {
                enabled: true
            },
            client: {
                enabled: true
            },
            reactQuery: {
                enabled: true
            }
        }
    });
    await builder.build();
});

it('builder default', async () => {
    const outPath = OUTPUT_DIR + '/default/';
    const s = (str: string) => FIXTURE_DIR + str;
    await codegen({
        contracts: [
            s('/vectis/factory'),
            s('/minter'),
            s('/daodao/cw-admin-factory'),
            s('/daodao/cw-code-id-registry'),
            {
                name: 'CwSingle',
                dir: s('/daodao/cw-proposal-single')
            }
        ],
        outPath,
        options: {
            bundle: {
                bundleFile: 'index.ts',
                scope: 'smart.contracts'
            },
            useShorthandCtor: false,
            types: {
                enabled: true
            },
            client: {
                enabled: true,
                execExtendsQuery: true
            },
            reactQuery: {
                enabled: true
            },
            recoil: {
                enabled: true
            },
            messageComposer: {
                enabled: true
            },
            messageBuilder: {
                enabled: true
            },
            useContractsHooks: {
                enabled: true,
            }
        }
    });
});

it('builder no extends', async () => {
    const outPath = OUTPUT_DIR + '/no-extends/';
    const s = (str: string) => FIXTURE_DIR + str;
    await codegen({
        contracts: [
            s('/vectis/factory'),
            s('/minter'),
            s('/daodao/cw-admin-factory'),
            s('/daodao/cw-code-id-registry'),
            {
                name: 'CwSingle',
                dir: s('/daodao/cw-proposal-single')
            }
        ],
        outPath,
        options: {
            bundle: {
                bundleFile: 'index.ts',
                scope: 'smart.contracts'
            },
            types: {
                enabled: true
            },
            client: {
                enabled: true,
                execExtendsQuery: false
            },
            reactQuery: {
                enabled: true
            },
            recoil: {
                enabled: true
            },
            messageComposer: {
                enabled: true
            },
            messageBuilder: {
                enabled: true
            }
        }
    });
});

it('builder set bundler path', async () => {
    const outPath = OUTPUT_DIR + '/bundler_test/contracts';
    const bundlerPath = OUTPUT_DIR + '/bundler_test/';
    const s = (str: string) => FIXTURE_DIR + str;
    await codegen({
        contracts: [
            s('/vectis/factory'),
            s('/minter'),
            s('/daodao/cw-admin-factory'),
            s('/daodao/cw-code-id-registry'),
            {
                name: 'CwSingle',
                dir: s('/daodao/cw-proposal-single')
            }
        ],
        outPath,
        options: {
            bundle: {
                bundlePath: bundlerPath,
                bundleFile: 'index.ts',
                scope: 'smart.contracts'
            },
            types: {
                enabled: true
            },
            client: {
                enabled: true,
                execExtendsQuery: false
            },
            reactQuery: {
                enabled: true
            },
            recoil: {
                enabled: true
            },
            messageComposer: {
                enabled: true
            },
            useContractsHooks: {
                enabled: true,
            }
        }
    });
});

