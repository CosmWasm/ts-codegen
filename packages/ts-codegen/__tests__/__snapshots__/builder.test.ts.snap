// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options tsClient.enabled 1`] = `
TSBuilder {
  "files": Array [],
  "options": Object {
    "abstractApp": Object {
      "clientPrefix": "App",
      "enabled": false,
    },
    "bundle": Object {
      "bundleFile": "bundle.ts",
      "enabled": true,
      "scope": "contracts",
    },
    "client": Object {
      "enabled": true,
      "execExtendsQuery": true,
      "noImplicitOverride": false,
    },
    "enabled": true,
    "messageComposer": Object {
      "enabled": false,
    },
    "msgBuilder": Object {
      "enabled": false,
    },
    "reactQuery": Object {
      "camelize": true,
      "enabled": true,
      "mutations": false,
      "optionalClient": false,
      "queryKeys": false,
      "version": "v3",
    },
    "recoil": Object {
      "enabled": false,
    },
    "types": Object {
      "aliasExecuteMsg": false,
      "enabled": true,
    },
  },
  "plugins": Array [
    TypesPlugin {
      "option": Object {
        "abstractApp": Object {
          "clientPrefix": "App",
          "enabled": false,
        },
        "bundle": Object {
          "bundleFile": "bundle.ts",
          "enabled": true,
          "scope": "contracts",
        },
        "client": Object {
          "enabled": true,
          "execExtendsQuery": true,
          "noImplicitOverride": false,
        },
        "enabled": true,
        "messageComposer": Object {
          "enabled": false,
        },
        "msgBuilder": Object {
          "enabled": false,
        },
        "reactQuery": Object {
          "camelize": true,
          "enabled": true,
          "mutations": false,
          "optionalClient": false,
          "queryKeys": false,
          "version": "v3",
        },
        "recoil": Object {
          "enabled": false,
        },
        "types": Object {
          "aliasExecuteMsg": false,
          "enabled": true,
        },
      },
      "utils": undefined,
    },
    ClientPlugin {
      "option": Object {
        "abstractApp": Object {
          "clientPrefix": "App",
          "enabled": false,
        },
        "bundle": Object {
          "bundleFile": "bundle.ts",
          "enabled": true,
          "scope": "contracts",
        },
        "client": Object {
          "enabled": true,
          "execExtendsQuery": true,
          "noImplicitOverride": false,
        },
        "enabled": true,
        "messageComposer": Object {
          "enabled": false,
        },
        "msgBuilder": Object {
          "enabled": false,
        },
        "reactQuery": Object {
          "camelize": true,
          "enabled": true,
          "mutations": false,
          "optionalClient": false,
          "queryKeys": false,
          "version": "v3",
        },
        "recoil": Object {
          "enabled": false,
        },
        "types": Object {
          "aliasExecuteMsg": false,
          "enabled": true,
        },
      },
      "utils": undefined,
    },
    AbstractAppPlugin {
      "option": Object {
        "abstractApp": Object {
          "clientPrefix": "App",
          "enabled": false,
        },
        "bundle": Object {
          "bundleFile": "bundle.ts",
          "enabled": true,
          "scope": "contracts",
        },
        "client": Object {
          "enabled": true,
          "execExtendsQuery": true,
          "noImplicitOverride": false,
        },
        "enabled": true,
        "messageComposer": Object {
          "enabled": false,
        },
        "msgBuilder": Object {
          "enabled": false,
        },
        "reactQuery": Object {
          "camelize": true,
          "enabled": true,
          "mutations": false,
          "optionalClient": false,
          "queryKeys": false,
          "version": "v3",
        },
        "recoil": Object {
          "enabled": false,
        },
        "types": Object {
          "aliasExecuteMsg": false,
          "enabled": true,
        },
      },
      "utils": undefined,
    },
    MessageComposerPlugin {
      "option": Object {
        "abstractApp": Object {
          "clientPrefix": "App",
          "enabled": false,
        },
        "bundle": Object {
          "bundleFile": "bundle.ts",
          "enabled": true,
          "scope": "contracts",
        },
        "client": Object {
          "enabled": true,
          "execExtendsQuery": true,
          "noImplicitOverride": false,
        },
        "enabled": true,
        "messageComposer": Object {
          "enabled": false,
        },
        "msgBuilder": Object {
          "enabled": false,
        },
        "reactQuery": Object {
          "camelize": true,
          "enabled": true,
          "mutations": false,
          "optionalClient": false,
          "queryKeys": false,
          "version": "v3",
        },
        "recoil": Object {
          "enabled": false,
        },
        "types": Object {
          "aliasExecuteMsg": false,
          "enabled": true,
        },
      },
      "utils": undefined,
    },
    ReactQueryPlugin {
      "option": Object {
        "abstractApp": Object {
          "clientPrefix": "App",
          "enabled": false,
        },
        "bundle": Object {
          "bundleFile": "bundle.ts",
          "enabled": true,
          "scope": "contracts",
        },
        "client": Object {
          "enabled": true,
          "execExtendsQuery": true,
          "noImplicitOverride": false,
        },
        "enabled": true,
        "messageComposer": Object {
          "enabled": false,
        },
        "msgBuilder": Object {
          "enabled": false,
        },
        "reactQuery": Object {
          "camelize": true,
          "enabled": true,
          "mutations": false,
          "optionalClient": false,
          "queryKeys": false,
          "version": "v3",
        },
        "recoil": Object {
          "enabled": false,
        },
        "types": Object {
          "aliasExecuteMsg": false,
          "enabled": true,
        },
      },
      "utils": undefined,
    },
    RecoilPlugin {
      "option": Object {
        "abstractApp": Object {
          "clientPrefix": "App",
          "enabled": false,
        },
        "bundle": Object {
          "bundleFile": "bundle.ts",
          "enabled": true,
          "scope": "contracts",
        },
        "client": Object {
          "enabled": true,
          "execExtendsQuery": true,
          "noImplicitOverride": false,
        },
        "enabled": true,
        "messageComposer": Object {
          "enabled": false,
        },
        "msgBuilder": Object {
          "enabled": false,
        },
        "reactQuery": Object {
          "camelize": true,
          "enabled": true,
          "mutations": false,
          "optionalClient": false,
          "queryKeys": false,
          "version": "v3",
        },
        "recoil": Object {
          "enabled": false,
        },
        "types": Object {
          "aliasExecuteMsg": false,
          "enabled": true,
        },
      },
      "utils": Object {
        "selectorFamily": "recoil",
      },
    },
    MsgBuilderPlugin {
      "option": Object {
        "abstractApp": Object {
          "clientPrefix": "App",
          "enabled": false,
        },
        "bundle": Object {
          "bundleFile": "bundle.ts",
          "enabled": true,
          "scope": "contracts",
        },
        "client": Object {
          "enabled": true,
          "execExtendsQuery": true,
          "noImplicitOverride": false,
        },
        "enabled": true,
        "messageComposer": Object {
          "enabled": false,
        },
        "msgBuilder": Object {
          "enabled": false,
        },
        "reactQuery": Object {
          "camelize": true,
          "enabled": true,
          "mutations": false,
          "optionalClient": false,
          "queryKeys": false,
          "version": "v3",
        },
        "recoil": Object {
          "enabled": false,
        },
        "types": Object {
          "aliasExecuteMsg": false,
          "enabled": true,
        },
      },
      "utils": undefined,
    },
  ],
}
`;

exports[`options undefined 1`] = `
TSBuilder {
  "files": Array [],
  "options": Object {
    "abstractApp": Object {
      "clientPrefix": "App",
      "enabled": false,
    },
    "bundle": Object {
      "bundleFile": "bundle.ts",
      "enabled": true,
      "scope": "contracts",
    },
    "client": Object {
      "enabled": true,
      "execExtendsQuery": true,
      "noImplicitOverride": false,
    },
    "enabled": true,
    "messageComposer": Object {
      "enabled": false,
    },
    "msgBuilder": Object {
      "enabled": false,
    },
    "reactQuery": Object {
      "camelize": true,
      "enabled": false,
      "mutations": false,
      "optionalClient": false,
      "queryKeys": false,
      "version": "v3",
    },
    "recoil": Object {
      "enabled": false,
    },
    "types": Object {
      "aliasExecuteMsg": false,
      "enabled": true,
    },
  },
  "plugins": Array [
    TypesPlugin {
      "option": Object {
        "abstractApp": Object {
          "clientPrefix": "App",
          "enabled": false,
        },
        "bundle": Object {
          "bundleFile": "bundle.ts",
          "enabled": true,
          "scope": "contracts",
        },
        "client": Object {
          "enabled": true,
          "execExtendsQuery": true,
          "noImplicitOverride": false,
        },
        "enabled": true,
        "messageComposer": Object {
          "enabled": false,
        },
        "msgBuilder": Object {
          "enabled": false,
        },
        "reactQuery": Object {
          "camelize": true,
          "enabled": false,
          "mutations": false,
          "optionalClient": false,
          "queryKeys": false,
          "version": "v3",
        },
        "recoil": Object {
          "enabled": false,
        },
        "types": Object {
          "aliasExecuteMsg": false,
          "enabled": true,
        },
      },
      "utils": undefined,
    },
    ClientPlugin {
      "option": Object {
        "abstractApp": Object {
          "clientPrefix": "App",
          "enabled": false,
        },
        "bundle": Object {
          "bundleFile": "bundle.ts",
          "enabled": true,
          "scope": "contracts",
        },
        "client": Object {
          "enabled": true,
          "execExtendsQuery": true,
          "noImplicitOverride": false,
        },
        "enabled": true,
        "messageComposer": Object {
          "enabled": false,
        },
        "msgBuilder": Object {
          "enabled": false,
        },
        "reactQuery": Object {
          "camelize": true,
          "enabled": false,
          "mutations": false,
          "optionalClient": false,
          "queryKeys": false,
          "version": "v3",
        },
        "recoil": Object {
          "enabled": false,
        },
        "types": Object {
          "aliasExecuteMsg": false,
          "enabled": true,
        },
      },
      "utils": undefined,
    },
    AbstractAppPlugin {
      "option": Object {
        "abstractApp": Object {
          "clientPrefix": "App",
          "enabled": false,
        },
        "bundle": Object {
          "bundleFile": "bundle.ts",
          "enabled": true,
          "scope": "contracts",
        },
        "client": Object {
          "enabled": true,
          "execExtendsQuery": true,
          "noImplicitOverride": false,
        },
        "enabled": true,
        "messageComposer": Object {
          "enabled": false,
        },
        "msgBuilder": Object {
          "enabled": false,
        },
        "reactQuery": Object {
          "camelize": true,
          "enabled": false,
          "mutations": false,
          "optionalClient": false,
          "queryKeys": false,
          "version": "v3",
        },
        "recoil": Object {
          "enabled": false,
        },
        "types": Object {
          "aliasExecuteMsg": false,
          "enabled": true,
        },
      },
      "utils": undefined,
    },
    MessageComposerPlugin {
      "option": Object {
        "abstractApp": Object {
          "clientPrefix": "App",
          "enabled": false,
        },
        "bundle": Object {
          "bundleFile": "bundle.ts",
          "enabled": true,
          "scope": "contracts",
        },
        "client": Object {
          "enabled": true,
          "execExtendsQuery": true,
          "noImplicitOverride": false,
        },
        "enabled": true,
        "messageComposer": Object {
          "enabled": false,
        },
        "msgBuilder": Object {
          "enabled": false,
        },
        "reactQuery": Object {
          "camelize": true,
          "enabled": false,
          "mutations": false,
          "optionalClient": false,
          "queryKeys": false,
          "version": "v3",
        },
        "recoil": Object {
          "enabled": false,
        },
        "types": Object {
          "aliasExecuteMsg": false,
          "enabled": true,
        },
      },
      "utils": undefined,
    },
    ReactQueryPlugin {
      "option": Object {
        "abstractApp": Object {
          "clientPrefix": "App",
          "enabled": false,
        },
        "bundle": Object {
          "bundleFile": "bundle.ts",
          "enabled": true,
          "scope": "contracts",
        },
        "client": Object {
          "enabled": true,
          "execExtendsQuery": true,
          "noImplicitOverride": false,
        },
        "enabled": true,
        "messageComposer": Object {
          "enabled": false,
        },
        "msgBuilder": Object {
          "enabled": false,
        },
        "reactQuery": Object {
          "camelize": true,
          "enabled": false,
          "mutations": false,
          "optionalClient": false,
          "queryKeys": false,
          "version": "v3",
        },
        "recoil": Object {
          "enabled": false,
        },
        "types": Object {
          "aliasExecuteMsg": false,
          "enabled": true,
        },
      },
      "utils": undefined,
    },
    RecoilPlugin {
      "option": Object {
        "abstractApp": Object {
          "clientPrefix": "App",
          "enabled": false,
        },
        "bundle": Object {
          "bundleFile": "bundle.ts",
          "enabled": true,
          "scope": "contracts",
        },
        "client": Object {
          "enabled": true,
          "execExtendsQuery": true,
          "noImplicitOverride": false,
        },
        "enabled": true,
        "messageComposer": Object {
          "enabled": false,
        },
        "msgBuilder": Object {
          "enabled": false,
        },
        "reactQuery": Object {
          "camelize": true,
          "enabled": false,
          "mutations": false,
          "optionalClient": false,
          "queryKeys": false,
          "version": "v3",
        },
        "recoil": Object {
          "enabled": false,
        },
        "types": Object {
          "aliasExecuteMsg": false,
          "enabled": true,
        },
      },
      "utils": Object {
        "selectorFamily": "recoil",
      },
    },
    MsgBuilderPlugin {
      "option": Object {
        "abstractApp": Object {
          "clientPrefix": "App",
          "enabled": false,
        },
        "bundle": Object {
          "bundleFile": "bundle.ts",
          "enabled": true,
          "scope": "contracts",
        },
        "client": Object {
          "enabled": true,
          "execExtendsQuery": true,
          "noImplicitOverride": false,
        },
        "enabled": true,
        "messageComposer": Object {
          "enabled": false,
        },
        "msgBuilder": Object {
          "enabled": false,
        },
        "reactQuery": Object {
          "camelize": true,
          "enabled": false,
          "mutations": false,
          "optionalClient": false,
          "queryKeys": false,
          "version": "v3",
        },
        "recoil": Object {
          "enabled": false,
        },
        "types": Object {
          "aliasExecuteMsg": false,
          "enabled": true,
        },
      },
      "utils": undefined,
    },
  ],
}
`;
